components/clicktime/clicktime.app.mjs

```diff
@@ -0,0 +1,190 @@
+      description: "",
+    },
+    isEligibleTimeOffAllocation: {
+      type: "boolean",
+      label: "Eligible for Time Off Allocation",
+      description: "Determines if the entity is eligible for time-off allocation",
+    },
+    name: {
+      type: "string",
+      label: "Name",
+      description: "",
+    },
+    notes: {
+      type: "string",
+      label: "Notes",
+      description: "",
+    },
+    shortName: {
+      type: "string",
+      label: "Short Name",
+      description: "A shorter version of the name for easier reference",
+    },
+    clientId: {
+      type: "string",
+      label: "Client ID",
+      description: "Unique identifier of the client associated with a job",
+      async options() {
+        const response = await this.getClients();
+        const clientIds = response.data;
+        return clientIds.map(({
+          ID, Name,
+        }) => ({
+          label: Name,
+          value: ID,
+        }));
+      },
+    },
+    endDate: {
+      type: "string",
+      label: "End Date",
+      description: "The end date of the job, i.e.: `2020-01-01`",
+    },
+    includeInRm: {
+      type: "boolean",
+      label: "Include in RM",
+      description: "Specifies whether the job should be included in RM",
+    },
+    isBillable: {
+      type: "boolean",
+      label: "Is Billable",
+      description: "Indicates whether the job is billable",
+    },
+    jobNumber: {
+      type: "string",
+      label: "Job Number",
+      description: "Unique identifier assigned to a job for tracking purposes",
+    },
+    startDate: {
+      type: "string",
+      label: "Start Date",
+      description: "",
+    },
+    timeRequiresApproval: {
+      type: "boolean",
+      label: "Time Requires Approval",
+      description: "Indicates whether time entries for the job require approval",
+    },
+    useCompanyBillingRate: {
+      type: "boolean",
+      label: "Use Company Billing Rate",
+      description: "Specifies whether the job uses the company-wide billing rate instead of a custom one",
+    },
+    costModel: {
+      type: "string",
+      label: "Cost Model",
+      description: "Defines the cost model used for the user",
+      options: constants.COST_MODELS,
+    },
+    costRate: {
+      type: "string",
+      label: "Cost Rate",
+      description: "The internal cost rate associated with the user",
+    },
+    email: {
+      type: "string",
+      label: "Email",
+      description: "The email address associated with the user",
+    },
+    employmentTypeId: {
+      type: "string",
+      label: "Employment Type ID",
+      description: "Unique identifier for the user's employment type",
+      async options() {
+        const response = await this.getEmploymentTypes();
+        const employmentTypeIds = response.data;
+        return employmentTypeIds.map(({
+          ID, Name,
+        }) => ({
+          label: Name,
+          value: ID,
+        }));
+      },
+    },
```

Comment:
Should use `limit` and `offset` to paginate employment types. https://developer.clicktime.com/docs/api/#/EmploymentTypes/ListEmploymentTypes